README.md

```diff
@@ -1296,6 +1299,156 @@ CREATE VIEW animal AS
 ```
 At this point, the `animal` view can be used in the SQLAlchemy Table for the purposes of compiling.
 
+## Macro System
+
+### Macro Edges
+Macro edges define a new edge in the schema by specifying how to compute it.
+Macro edges are similar to views in SQL. However, there's a few differences:
+- Macros are stored in a macro registry, that does not necessarily live inside the database. This has
+two avantages:
+  - Macros can be created locally by clients without database access.
+  - Macros work even on schemas that have been stitched from the schemas of multiple backends.
+- When a query with macros is compiled, a preprocessing step generates an equivalent query without
+macros by expanding.
+
+As an example, consider the following query:
+```graphql
+{
+    Animal {
+        name @output(out_name: "animal_name")
+        out_Animal_ParentOf {
+            out_Animal_ParentOf {
+                name @output(out_name: "grandchild_name")
+            }
+        }
+    }
+}
+```
+Let's introduce the `out_Animal_GrandparentOf` macro to simplify our query:
+```graphql
+{
+    Animal @macro_edge_definition(name: "out_Animal_GrandparentOf") {
+        out_Animal_ParentOf {
+            out_Animal_ParentOf @macro_edge_target {
+                uuid
+            }
+        }
+    }
+}
+```
+The directive `@macro_edge_definition` declares that this is a macro, and gives it a name.
+To use it as any other edge, we will refer to it by the name given here.
+The directive `@macro_edge_target` specifies where the grandchild of the animal is.
+Now let's use this macro to rewrite our query:
+```graphql
+{
+    Animal {
+        name @output(out_name: "animal_name")
+        out_Animal_GrandparentOf {
+            name @output(out_name: "grandchild_name")
+        }
+    }
+}
+```
+Macro edges can make use of tags, filters, etc, to define more complicated macros like this one
+```graphql
+Animal @macro_edge_definition(name: "out_Animal_RichYoungerSiblings") {
+    net_worth @tag(tag_name: "net_worth")
+    out_Animal_BornAt {
+        event_date @tag(tag_name: "birthday")
+    }
+    in_Animal_ParentOf {
+        out_Animal_ParentOf @macro_edge_target {
+            net_worth @filter(op_name: ">", value: ["%net_worth"])
+            out_Animal_BornAt {
+                event_date @filter(op_name: "<", value: ["%birthday"])
+            }
+        }
+    }
+}
+```
+Macro edges can also use runtime parameters. The values for the parameters are provided when
+the macro is registered, and not when it is used. See the section below to see how to use
```

Comment:
See [Using the macro registry](#using-the-macro-registry) for an example of how to use macros with runtime parameters.